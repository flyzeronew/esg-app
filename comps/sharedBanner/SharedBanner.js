import styles from './SharedBanner.module.css'
function SharedBanner({title, description}) {
    return (        
        <div className={styles.sharedBanner}>
            <div className={styles.mask}></div>
            <div className={styles.box}>
                <h1 className={styles.title}>{title}</h1>
                <div className={styles.txt}>
                    <p>{description}</p>
                    <div className={styles.line}></div>
                </div>
            </div>
        </div>
    )
}
export default SharedBanner;