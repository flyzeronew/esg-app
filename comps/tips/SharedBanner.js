import React, { Component } from "react"

function SharedBanner(props) {
    const appUrl = process.env.APP_URL;
    return (        
        <div className="sharedBanner">
            <div className="mask"></div>
            <div className="box">
                <h1 className="title">永續生活小撇步</h1>
                <div className="txt">
                    <p>踏上永續生活的路上，小小的舉措，我們珍愛這片土地，永續生活於其中。</p>
                    <div className="line"></div>
                </div>
            </div>
        </div>
    )
}
export default SharedBanner;