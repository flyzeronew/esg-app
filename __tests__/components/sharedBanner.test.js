import { render } from '@testing-library/react';
import SharedBanner from './../../comps/sharedBanner/SharedBanner';

describe('Shared Banner Component', () => {
    const mockBanner = {
        title: "Here is ESG",
        description: "This is were the description comes up"
    }
    it("renders Banner unchanged", () => {
        const { container } = render(<SharedBanner {...mockBanner} />);
        expect(container).toMatchSnapshot();
    });
    it('renders footer component with correct class', () => {
        const { getByText } = render(<SharedBanner {...mockBanner} />);
        const bannerTitle = getByText(mockBanner.title);
        const bannerDescription = getByText(mockBanner.description);
        expect(bannerTitle).toBeInTheDocument();
        expect(bannerDescription).toBeInTheDocument();
    });
});
