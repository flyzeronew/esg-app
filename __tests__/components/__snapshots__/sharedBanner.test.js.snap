// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Banner Component renders Banner unchanged 1`] = `
<div>
  <div
    class="sharedBanner"
  >
    <div
      class="mask"
    />
    <div
      class="box"
    >
      <h1
        class="title"
      >
        Here is ESG
      </h1>
      <div
        class="txt"
      >
        <p>
          This is were the description comes up
        </p>
        <div
          class="line"
        />
      </div>
    </div>
  </div>
</div>
`;
